include/session/config/groups/members.h

```diff
@@ -55,7 +55,11 @@ LIBSESSION_EXPORT int groups_members_init(
         const unsigned char* ed25519_secretkey,
         const unsigned char* dump,
         size_t dumplen,
-        char* error) __attribute__((warn_unused_result));
+        char* error)
+#ifdef __GNUC__
+        __attribute__((warn_unused_result))
+#endif
+        ;
```

Comment:
There are a bunch more of these in other C headers.  It would be nicer to add this into `export.h` (which gets included from all the C headers):
```C++
#ifdef __GNUC__
#define LIBSESSION_WARN_UNUSED __attribute__((warn_unused_result))
#else
#define LIBSESSION_WARN_UNUSED
#endif
```
and then we can just `__attribute__((warn_unused_result));` -> `LIBSESSION_WARN_UNUSED;` everywhere rather than needing a few dozen ifdefs.

(I count a couple dozen in session/config/various*.h, and one in session/network.h)

Reply:
should be fixed (but I don't have a network.h on my current branch)

Reply:
Ah, right, that is from the libquic branch.